src/coreComponents/meshUtilities/InternalWellboreGenerator.cpp

```diff
@@ -0,0 +1,216 @@
+    setApplyDefaultValue( 360.0 ).
+    setInputFlag( InputFlags::OPTIONAL ).
+    setDescription( "Tangent angle defining geometry size: 90 for quarter, 180 for half and 360 for full wellbore geometry" );
+
+  registerWrapper( viewKeyStruct::rOutString(), &m_rOut ).
+    setApplyDefaultValue( 1.0 ).
+    setInputFlag( InputFlags::OPTIONAL ).
+    setDescription( "Farfield distance from wellbore center" );
+
+  registerWrapper( viewKeyStruct::rElemsString(), &m_rElems ).
+    setApplyDefaultValue( 10 ).
+    setSizedFromParent( 0 ).
+    setInputFlag( InputFlags::OPTIONAL ).
+    setDescription( "Number of elements in the radial direction" );
+
+  registerWrapper( viewKeyStruct::tElemsString(), &m_tElems ).
+    setApplyDefaultValue( 40 ).
+    setSizedFromParent( 0 ).
+    setInputFlag( InputFlags::OPTIONAL ).
+    setDescription( "Number of elements in the tangent direction" );
+
+  registerWrapper( viewKeyStruct::rBiasString(), &m_rBias ).
+    setApplyDefaultValue( -0.8 ).
+    setSizedFromParent( 0 ).
+    setInputFlag( InputFlags::OPTIONAL ).
+    setDescription( "Bias of element sizes in the radial direction" );
+
+  registerWrapper( viewKeyStruct::trajectoryString(), &m_trajectory ).
+    setSizedFromParent( 0 ).
+    setInputFlag( InputFlags::OPTIONAL ).
+    setDescription( "Coordinates defining the wellbore trajectory" );
+}
+
+void InternalWellboreGenerator::postProcessInput()
+{
+  m_vertices[0].resize( 2 );
+  m_vertices[1].resize( 2 );
+  m_vertices[2].resize( 2 );
+  m_nElems[0].resize( 1 );
+  m_nElems[1].resize( 1 );
+  m_nElemBias[0].resize( 1 );
+
+  m_vertices[0][0]  = m_radius;
+  m_vertices[0][1]  = m_rOut;
+  m_vertices[1][0]  = 0;
+  m_vertices[1][1]  = m_theta;
+  m_vertices[2][0]  = m_trajectory[0][2];
+  m_vertices[2][1]  = m_trajectory[1][2];
+  m_nElems[0][0]    = m_rElems;
+  m_nElems[1][0]    = m_tElems;
+  m_nElemBias[0][0] = m_rBias;
+
+  InternalMeshGenerator::postProcessInput();
+}
+
+void InternalWellboreGenerator::generateMesh( DomainPartition & domain )
+{
+  InternalMeshGenerator::generateMesh( domain );
+
+  Group & meshBodies = domain.getGroup( string( "MeshBodies" ));
+  MeshBody & meshBody = meshBodies.registerGroup< MeshBody >( this->getName() );
+  MeshLevel & meshLevel0 = meshBody.registerGroup< MeshLevel >( string( "Level0" ));
+  NodeManager & nodeManager = meshLevel0.getNodeManager();
+  Group & nodeSets = nodeManager.sets();
+
+  // Wellbore nodesets
+  SortedArray< localIndex > & rnegNodes = nodeSets.registerWrapper< SortedArray< localIndex > >( string( "rneg" ) ).reference();
+  SortedArray< localIndex > & rposNodes = nodeSets.registerWrapper< SortedArray< localIndex > >( string( "rpos" ) ).reference();
+  SortedArray< localIndex > & tnegNodes = nodeSets.registerWrapper< SortedArray< localIndex > >( string( "tneg" ) ).reference();
+  SortedArray< localIndex > & tposNodes = nodeSets.registerWrapper< SortedArray< localIndex > >( string( "tpos" ) ).reference();
```

Comment:
I realize that those variable are indeed part of the user interface of your feature (people will ask for these specific names in their input files).
Could you document above that
```c++
// rneg, rpos, tneg and tpos are the named used by the end-used in the input files. Consider modifying them with care.
```

Reply:
done